MusicStore.Business/Services/MusicStoreDisplayService.cs

```diff
@@ -0,0 +1,152 @@
+            {
+                throw new ArgumentNullException(nameof(albumsList), $"{nameof(albumsList)} less then 0");
+            }
+
+            var resultOfPagination = _albumRepository.MakePagination(albumsList.ToList(), page);
+
+            if (resultOfPagination == null)
+            {
+                throw new ArgumentNullException(nameof(resultOfPagination), $"{nameof(resultOfPagination)} less then 0");
+            }
+
+            IndexViewItem<Domain.DataTransfer.Album> indexViewDomainAlbums = new IndexViewItem<Domain.DataTransfer.Album>();
+            indexViewDomainAlbums.PageInfo = new Repository<Domain.DataTransfer.Album>();
+            indexViewDomainAlbums.PageInfo.PageNumber = resultOfPagination.PageInfo.PageNumber;
+            indexViewDomainAlbums.PageInfo.PageSize = resultOfPagination.PageInfo.PageSize;
+            indexViewDomainAlbums.PageInfo.TotalItems = resultOfPagination.PageInfo.TotalItems;
+            indexViewDomainAlbums.Items = resultOfPagination.Items.Select(_mapAlbum.AutoMap);
+            return indexViewDomainAlbums;
+        }
+
+        public IList<Domain.DataTransfer.Song> GetSongsListFromAlbum(int albumId)
+        {
+            if (albumId < 0)
+            {
+                throw new ArgumentException("userId < 0 in musicStoreService DisplayAllAvailableSongs", nameof(albumId));
+            }
+
+            var album = _albumRepository.GetItem(albumId);
+
+            if(album == null || album.Songs == null)
+            {
+                throw new Exception("album is null or album.Songs is null");
+            }
+
+            List<Domain.DataTransfer.Song> domainSongsList = new List<Domain.DataTransfer.Song>();
+            domainSongsList.AddRange(album.Songs.Select(_mapSong.AutoMap));
+            return domainSongsList;
+        }
+
+        public IList<Domain.DataTransfer.Song> GetSongsListFromAlbumAvailableForBuyByUser(int albumId, int userId)
+        {
+            if (albumId < 0)
+            {
+                throw new ArgumentException($"{nameof(albumId)} < 0 in musicStoreDisplayService GetSongsListAvailableForBuyByUser", nameof(albumId));
+            }
+
+            if (userId < 0)
+            {
+                throw new ArgumentException($"{nameof(userId)} < 0 in musicStoreDisplayService GetSongsListAvailableForBuyByUser", nameof(userId));
+            }
+
+            var album = _albumRepository.GetItem(albumId);
+            var user = _userRepository.GetItem(userId);
+
+            if (album == null || album.Songs == null)
+            {
+                throw new Exception("album is null or album.Songs is null");
+            }
+
+            if (user == null || user.BoughtSongs == null)
+            {
+                throw new Exception("user is null or user.BoughtSongs is null");
+            }
+
+            List<Domain.DataTransfer.Song> domainSongsList = new List<Domain.DataTransfer.Song>();
+
+            foreach(var albumSong in album.Songs)
```

Comment:
Investigate does it possible to replace this code with LINQ

Reply:
not done(it is comment for me)